qiskit_experiments/characterization/qubit_spectroscopy.py

```diff
@@ -369,14 +366,41 @@ def __init__(
         if len(frequencies) < 3:
             raise QiskitError("Spectroscopy requires at least three frequencies.")
 
-        if unit not in self.__units__:
-            raise QiskitError(f"Unsupported unit: {unit}.")
+        if unit == "Hz":
+            self._frequencies = frequencies
+        else:
+            self._frequencies = [apply_prefix(freq, unit) for freq in frequencies]
 
-        self._frequencies = [freq * self.__units__[unit] for freq in frequencies]
         self._absolute = absolute
 
         super().__init__([qubit])
 
+    def _schedule(self, backend: Optional[Backend] = None) -> Tuple[pulse.ScheduleBlock, Parameter]:
+        """Create the spectroscopy schedule."""
+        freq_param = Parameter("frequency")
+        with pulse.build(backend=backend, name="spectroscopy") as schedule:
+            pulse.set_frequency(freq_param, pulse.drive_channel(self.physical_qubits[0]))
```

Comment:
I think we should use `shift_frequency`. If we want to add some post circuits (actually it can because `_template_circuit` can be overridden), this affects all following gate operations.

Reply:
Sounds good. See here: 00d915f